extension.js

```diff
@@ -12,56 +12,63 @@ const GObject = imports.gi.GObject;
 function _get_lan_ip() {
     // Ask the IP stack what route would be used to reach 1.1.1.1 (Cloudflare DNS)
     // Specifically, what src would be used for the 1st hop?
-    var command_output_bytes = GLib.spawn_command_line_sync('ip route get 1.1.1.1')[1];
-    var command_output_string = '';
-
-    for (var current_character_index = 0;
-        current_character_index < command_output_bytes.length;
-        ++current_character_index)
-    {
-        var current_character = String.fromCharCode(command_output_bytes[current_character_index]);
+    const command_output_bytes = GLib.spawn_command_line_sync('ip route get 1.1.1.1')[1];
+    let command_output_string = '';
+
+    for (let current_character_index = 0;
+         current_character_index < command_output_bytes.length;
+         ++current_character_index) {
+        const current_character = String.fromCharCode(command_output_bytes[current_character_index]);
         command_output_string += current_character;
     }
 
     // Output of the "ip route" command will be a string
     // " ... src 1.2.3.4 ..."
     // So basically we want the next token (word) immediately after the "src"
     // word, and nothing else. This is considerd our LAN IP address.
-    var Re = new RegExp(/src [^ ]+/g);
-    var matches = command_output_string.match(Re);
-    var lanIpAddress;
+    const Re = new RegExp(/src [^ ]+/g);
+    const matches = command_output_string.match(Re);
+    let lanIpAddress = '';
     if (matches) {
         lanIpAddress = matches[0].split(' ')[1];
-    } else {
-        lanIpAddress = '';
     }
- 
     return lanIpAddress;
 }
 
+function copyToClipboard(text) {
+    St.Clipboard.get_default().set_text(St.ClipboardType.CLIPBOARD, text);
+}
+
 // Our PanelMenu.Button subclass
-var LanIPAddressIndicator = class LanIPAddressIndicator extends PanelMenu.Button {
+let LanIPAddressIndicator = class LanIPAddressIndicator extends PanelMenu.Button {
 
     _init() {
         // Chaining up to the super-class
         super._init(0.0, "LAN IP Address Indicator", false);
-        
+
         this.buttonText = new St.Label({
             text: 'Loading...',
             y_align: Clutter.ActorAlign.CENTER
         });
         this.add_child(this.buttonText);
+
+        this.connect('button_press_event', () => {
+            copyToClipboard(this.buttonText.get_text());
+        });
+
```

Comment:
This could use some logic to not copy things like "Loading..." or "" (an empty string) to the clipboard.